### PR TITLE
Add xdp device exclusion filters

### LINKED ISSUE #2968: Ignore XDP entries a la ['bad_if_regexp']?
Discovery times on segments with IP phones and access points that participate in CDP/LLDP are very high (end devices such as these outnumber infrastructure hardware by a factor of 50-to-1). The logic that's been written to ignore interface types and interface regex would ameliorate this. Can this be put on the radar as a 'nice to have?'

Thanks for all of the amazing work!

### PR DESCRIPTION
**Pull request by NetworkNub**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

Fixes #2968

Adds filters to exclude devices from being discovered via xdp. This increases discovery performance of devices when connected to non snmp capable devices like phones and APs.

### CODE REVIEW

File: includes/defaults.inc.php [hunk 1]
```diff
@@ -199,6 +199,9 @@
 
 // Autodiscovery Settings
 $config['autodiscovery']['xdp'] = true;
+$config['autodiscovery']['xdp_exclude']['sysdesc_regexp'][] = '/-K9W8-/'; // Cisco Lightweight Access Point
+$config['autodiscovery']['cdp_exclude']['platform_regexp'][] = '/^Cisco IP Phone/'; //Cisco IP Phone
```
Comment (laf) [inline includes/defaults.inc.php]: What's the reason why these are cdp specific rather than having them all just xdp?

Comment (NetworkNub) [inline includes/defaults.inc.php]: sysname and sysdesc are sent in both LLDP and CDP protocols so xdp was used for those two.

platform is sent only in the CDP protocol and has no corresponding entry in the LLDP protocol.

Here's what the same device looks like in both protocols:

```
[cdpCacheAddressType] => ip
[cdpCacheAddress] => 0A 0C 08 5A
[cdpCacheVersion] => Cisco IOS Software, C2600 Software (AP3G2-K9W8-M), Version 15.3(3)JA7, RELEASE SOFTWARE (fc2)
[cdpCacheDeviceId] => abc-102-ap1
[cdpCacheDevicePort] => GigabitEthernet0
[cdpCachePlatform] => cisco AIR-CAP2602I-A-K9
[cdpCacheCapabilities] => 00 00 00 26
[cdpCacheNativeVLAN] => 0
[cdpCacheDuplex] => fullduplex
[cdpCacheLastChange] => 79:10:46:04.79

[lldpRemChassisIdSubtype] => macAddress
[lldpRemChassisId] => 24 01 C7 F7 43 60
[lldpRemPortIdSubtype] => interfaceName
[lldpRemPortId] => Gi0
[lldpRemPortDesc] => GigabitEthernet0
[lldpRemSysName] => abc-102-ap1.xyz.k12.va.us
[lldpRemSysDesc] => Cisco IOS Software, C2600 Software (AP3G2-K9W8-M), Version 15.3(3)JA7, RELEASE SOFTWARE (fc2)
[lldpRemSysCapSupported] => 20 00
[lldpRemSysCapEnabled] => 20 00
[lldpRemManAddrIfSubtype] => systemPortNumber
[lldpRemManAddrIfId] => 0
[lldpRemManAddrOID] => zeroDotZero

```

**Comment by laf** [comment]
@librenms/reviewers  please review.

I've centralised the foreach calls to a function here. I'd have liked to have used str_contains() but we should and do need to log what's skipped so chosen to do another function to keep clean code.

@NetworkNub be good if you can test this. Also FYI I've removed the default excludes as we shouldn't be dictating what devices end users discovery or not. People are free to add these exclusions manually.

**Comment by NetworkNub** [comment]
@laf fixed two issues but looks good otherwise.  Tested on a Cisco switch running both CDP and LLDP. 

re Default excludes
Both Cisco phones and lightweight APs (sysdesc -K9W8-) don't support SNMP and are not discoverable. Cisco autonomous APs (sysdesc -K9W7-) do support SNMP and would continue to be discovered using those defaults.
The IP cameras support SNMP and I agree those should not be excluded by default.


On a switch with 24 LWAPs and one phone:
w/o filters: `>> Runtime for discovery module 'discovery-protocols': 2920. seconds`
w/ filters: `>> Runtime for discovery module 'discovery-protocols': 5.893 seconds`



Side note; LLDP link discovery seems to be broken on a Cisco 3750X switch both with or without these changes. snmpwalk lldpRemoteSystemsData returns data but the devices don't show up on the ports/neighbors page. I don't have any other vendors to test with. I will open a separate issue for this.

**Comment by laf** [comment]
I've re-added the defaults aside from the IP cams.

**Comment by jongaynor** [comment]
You guys are the best. Thanks much!

**Pull request merged**